front/app/containers/Admin/reporting/components/ReportBuilder/TopBar/index.tsx

```diff
@@ -76,6 +79,32 @@ const ContentBuilderTopBar = ({
     }
   };
 
+  useEffect(() => {
```

Comment:
The famous auto-save after a timeout

Reply:
So in the off chance you leave the page before 5 seconds are up it won't save? Think we can live with that?

Reply:
Yeah... I mean, we can make the timeout shorter but then we increase the chance that is saves before the template is fully loaded (which in the best case means saving an incomplete layout, and in the worst case an invalid layout that causes craftjs to crash). And I don't think people will leave that quickly, especially once we have Fran's modal/alert stuff that asks you to save before leaving the page